src/apis/impl/AuthenticationApiImpl.ts

```diff
@@ -100,10 +102,69 @@ export class AuthenticationApiImpl implements AuthenticationApi {
     return null
   }
 
-  async initApiAndUserAuthenticationToken(process: AuthenticationProcess, validationCode: string, tokenAndKeyPairProvider: (groupId: string, userId: string) => ([string, [string, string]] | undefined)): Promise<[MedTechApi, ApiInitialisationResult]> {
+  async authenticateAndAskForAccess(userLogin: string, shortLivedToken: string, dataOwnerKeyPair: [string, string] | undefined, tokenAndKeyPairProvider: (groupId: string, userId: string) => ([string, [string, string]] | undefined)): Promise<AuthenticationResult | null> {
+    const authenticationResult = await this.initUserTokenAndCrypto(userLogin, shortLivedToken, dataOwnerKeyPair, tokenAndKeyPairProvider);
+
+    const loggedUser = await authenticationResult.medTechApi.userApi.getLoggedUser();
+    if (!loggedUser) throw new Error("User log in failed");
+    if (!loggedUser.patientId) throw new Error("User is not associated to any Patient");
+
+    const patientUser = await authenticationResult.medTechApi.patientApi.getPatient(loggedUser.patientId);
```

Comment:
Same : You have a getDataOwner API you can use in the base sdk to treat patient / hcp & device. That way, the service will be working for each type of user